crypto/asn1/a_dup.c

```diff
@@ -48,13 +48,18 @@ void *ASN1_dup(i2d_of_void *i2d, d2i_of_void *d2i, const void *x)
 
 void *ASN1_item_dup(const ASN1_ITEM *it, const void *x)
 {
+    const ASN1_AUX *aux = it->funcs;
+    ASN1_aux_cb *asn1_cb = aux != NULL ? aux->asn1_cb : NULL;
     unsigned char *b = NULL;
     const unsigned char *p;
     long i;
-    void *ret;
+    ASN1_VALUE *ret;
 
     if (x == NULL)
         return NULL;
+    if (asn1_cb != NULL
+            && !asn1_cb(ASN1_OP_DUP_PRE, (ASN1_VALUE **)&x, it, NULL))
```

Comment:
I was thinking about this call just before falling asleep last night, and I wonder if it would be more consistent if called like this:

``` C
    if (asn1_cb != NULL
            && !asn1_cb(ASN1_OP_DUP_PRE, NULL, it, (void *)x))
```

Reply:
No problem, done.

Reply:
@DDvO  I dig and get the crash reason :

you add this line in your PR :
`const ASN1_AUX *aux = it->funcs;`

But for the X5009_NAME type, its respective `void* it->funcs` is not pointing to an expected `509_AUX` object, 

It is pointing to a structure named `x509_name_ff`, 
And so the code ends calling 'x509_name_ex_d2i' trough it.